### ISSUE TITLE
Add function to test structural changes in data.frame / linelist

### ISSUE THREAD

**Issue by thibautjombart** [reconhub/linelist#34]
## Diagnosing structural changes

- we have a reference object `x` which can be a `data.frame` or a `linelist`
- we have a new object `y` of the same type
- (optionally) we have  a vector of variable names which "matter", i.e. only test for that
- we want to test if objects are structurally similar, based on column names and types, with various diagnostics i.e.:
  
1. identical column names and types
2. as above, but different order
3. only some partial matches in column names
4. similarly named columns have different types

Ideally the ouput of this function would:
- contain enough for the second part (see below)
- be verbose and give a nice summary to the user of what has changed (or hasn't)

## Impose a reference structure on a dataset

With the above in mind, it would be great to be able to use this diagnostic so that the structure of `y` can be altered to match that of `x`, e.g.
```
restructure <- function(y, reference = x, ...) {
## essentially renaming and reordering variables 
...
}

## use:
restructure(y, reference = x, date_symptoms = "date_onset", toto = "tata")
```
where `...` captures renaming of variables (toto: col in `y`; tata: col in `x`), and optionally type conversion as a second argument, e.g.:
```
restructure(y, reference = x, 
                  date_symptoms = c("date_onset", "Date"), 
                  toto = "tata", case_id = c("id", "character"))
```

**Comment by zkamvar** [comment]
> where `...` captures renaming of variables (toto: col in `y`; tata: col in `x`), and optionally type conversion as a second argument, e.g.:
> 
> ```
> restructure(y, reference = x, 
>                   date_symptoms = c("date_onset", "Date"), 
>                   toto = "tata", case_id = c("id", "character"))
> ```

If you are using `x` as a reference, wouldn't it make sense just to make the columns of `y` match `x`?

**Comment by zkamvar** [comment]
I just realized that my jet-lagged brain has been trying to approach this problem completely wrong! I was assuming that you wanted a sort of diff between two data sets showing changes. Instead, you want to diagnose column order and type and then try to match them!

**Comment by zkamvar** [comment]
This was fixed in #50
